content/tutorials/build-a-streaming-data-source-plugin.md

```diff
@@ -0,0 +1,173 @@
+1. A line graph is rendered with one series consisting of two data points.
+1. Save the dashboard.
+
+### Troubleshooting
+
+#### Grafana doesn't load my plugin
+
+By default, Grafana requires backend plugins to be signed. To load unsigned backend plugins, you need to
+configure Grafana to [allow unsigned plugins](https://grafana.com/docs/grafana/latest/plugins/plugin-signature-verification/#allow-unsigned-plugins).
+For more information, refer to [Plugin signature verification](https://grafana.com/docs/grafana/latest/plugins/plugin-signature-verification/#backend-plugins).
+
+{{< /tutorials/step >}}
+{{< tutorials/step title="Anatomy of a backend plugin" >}}
+
+As you may notice till this moment we did the same steps described in [build a backend datasource plugin tutorial](https://grafana.com/tutorials/build-a-data-source-backend-plugin/). At this point, you should be familiar with backend plugin structure and a way how data querying and health check capabilities could be implemented. Let's take the next step and discuss how datasource plugin can handle data streaming.
+
+{{< /tutorials/step >}}
+{{< tutorials/step title="Add streaming capabilities" >}}
+
+What we want to achieve here is to issue a query to load initial data from a datasource plugin and then switching to data streaming mode where the plugin will push data frames to Grafana time-series panel.
+
+In short – implementing a streaming plugin means implementing a `backend.StreamHandler` interface which contains `SubscribeStream`, `RunStream`, and `PublishStream` methods.
+
+`SubscribeStream` is a method where the plugin has a chance to authorize user subscription requests to a channel. Users on the frontend side subscribe to different channels to consume real-time data.
+
+When returning a `data.Frame` with initial data we can return a special field `Channel` to let the frontend know that we are going to stream data frames after initial data load. When the frontend receives a frame with a `Channel` set it automatically issues a subscription request to that channel.
+
+In Grafana channel consists of 3 parts delimited by `/`:
+
+* Scope
+* Namespace
+* Path
+
+For example, the channel `grafana/dashboard/xyz` has the scope `grafana`, namespace `dashboard`, and path `xyz`.
+
+Scope, namespace and path can only have ASCII alphanumeric symbols (A-Z, a-z, 0-9), `_` (underscore) and `-` (dash) at the moment. The path part can additionally have `/`, `.` and `=` symbols. The meaning of scope, namespace and path is context-specific.
+
+Scope determines the purpose of a channel in Grafana. For datasource plugin channels Grafana uses `ds` scope. Namespace in the case of datasource channels is a datasource unique ID (UID) which is issued by Grafana at the moment of datasource creation. The path is a custom string that plugin authors free to choose themselves (just make sure it consists of allowed symbols). I.e. datasource channel looks like `ds/<DATASOURCE_UID>/<CUSTOM_PATH>`.
```

Comment:
This part we should incorporate and move to grafana core docs

Reply:
We should give a link to docs here instead of describing right?

Reply:
https://github.com/grafana/grafana/pull/34642

Reply:
When a docs source is available, link to it. That way, if the main docs information changes, the tutorial will remain useful.